usaspending_api/references/management/commands/load_object_classes.py

```diff
@@ -0,0 +1,227 @@
+from usaspending_api.common.helpers.sql_helpers import get_connection
+from usaspending_api.common.helpers.text_helpers import standardize_nullable_whitespace as prep
+from usaspending_api.common.helpers.timing_helpers import Timer
+from usaspending_api.references.models import ObjectClass
+
+
+OBJECT_CLASS_PATTERN = re.compile("[12]?[0-9]{3}")
+
+CREATE_TEMP_TABLE = """
+    drop table if exists temp_load_object_classes;
+
+    create temporary table temp_load_object_classes (
+        row_number int,
+        major_object_class text,
+        major_object_class_name text,
+        object_class text,
+        object_class_name text,
+        direct_reimbursable text,
+        direct_reimbursable_name text
+    );
+"""
+
+logger = logging.getLogger("console")
+
+RawObjectClass = namedtuple("RawObjectClass", ["row_number", "object_class", "object_class_name"])
+
+FullObjectClass = namedtuple(
+    "FullObjectClass",
+    [
+        "row_number",
+        "major_object_class",
+        "major_object_class_name",
+        "object_class",
+        "object_class_name",
+        "direct_reimbursable",
+        "direct_reimbursable_name",
+    ],
+)
+
+
+class Command(mixins.ETLMixin, BaseCommand):
+
+    help = "Load object class CSV file.  If anything fails, nothing gets saved.  DOES NOT DELETE RECORDS."
+    object_class_file = None
+    etl_logger_function = logger.info
+
+    def add_arguments(self, parser):
+
+        parser.add_argument(
+            "object_class_file", metavar="FILE", help="Path or URI of the raw object class CSV file to be loaded."
+        )
+
+    def handle(self, *args, **options):
+
+        self.object_class_file = options["object_class_file"]
+
+        logger.info("OBJECT CLASS FILE: {}".format(self.object_class_file))
+
+        with Timer("Load object classes"):
+
+            try:
+                with transaction.atomic():
+                    self._perform_load()
+                    t = Timer("Commit transaction")
+                    t.log_starting_message()
+                t.log_success_message()
+            except Exception:
+                logger.error("ALL CHANGES ROLLED BACK DUE TO EXCEPTION")
+                raise
+
+            try:
+                self._vacuum_tables()
+            except Exception:
+                logger.error("CHANGES WERE SUCCESSFULLY COMMITTED EVEN THOUGH VACUUMS FAILED")
+                raise
+
+    def _read_raw_object_classes_csv(self):
+
+        object_classes = read_csv_file_as_list_of_dictionaries(self.object_class_file)
+        if len(object_classes) < 1:
+            raise RuntimeError("Object class file '{}' appears to be empty".format(self.object_class_file))
+
+        self.raw_object_classes = [
+            RawObjectClass(
+                row_number=row_number,
+                object_class=prep(object_class["MAX OC Code"]),
+                object_class_name=prep(object_class["MAX Object Class name"]),
```

Comment:
Just to confirm, it is agreeable to convert all whitespace characters to spaces and strip trailing and leading whitespace here? (We've been dealing with the fallout of previous decisions like that)

Reply:
I think it would have been ok (famous last words) since the code must be a three or four digit value and the description is just a description, but I went ahead and split the difference with you and dropped back to trimming only leading and trailing spaces.

This is designed to rectify a common problem with saw with the agency csv whereby something kept adding spaces to fields (almost had to be Excel since it is the primary tool used to maintain that file).